### PR TITLE
Changing occurrences of "neighbour" to EN-US spelling, "neighbor"

### LINKED ISSUE #6202: Parameter name 'neighbourhood' uses non-US spelling.
In the same vein as #2604, we may want to change parameter `neighbourhood` to `neighborhood` in function `measure.perimeter`: https://github.com/scikit-image/scikit-image/blob/55dee6421c1411388eb299f70244c90d956140ef/skimage/measure/_regionprops_utils.py#L186

_Originally posted by @alexdesiqueira in https://github.com/scikit-image/scikit-image/pull/6200#discussion_r790163993_:

@mkcor thanks for pointing that out. Yeah, I think a deprecation cycle would be good. Could you open a new issue based on that? I'll check the other places we would need to deprecate, also. We could take advantage of a new PR to change all of them at once.

### PR DESCRIPTION
**Pull request by alexdesiqueira**
## Description

Changing occurrences of "neighbourhood" to EN-US spelling, "neighborhood," and deprecating changes as necessary.
Closes #6202.

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by alexdesiqueira** [comment]
Fixed first round of occurrences presented at [skimage#6202:comment](https://github.com/scikit-image/scikit-image/issues/6202#issuecomment-1019382603). Changed some "neighbours" and similars around. Didn't deprecate any functions yet, though.
And then I though, "hey, let's check _neighbour_ instead and fix them all. Here's the list we have right now.

```bash
❯ rg neighbour
skimage/draw/draw_nd.py
58:    pixels in the line will be either direct or diagonal neighbours in

skimage/transform/_warps_cy.pyx
7:from .._shared.interpolation cimport (nearest_neighbour_interpolation,
166:        interp_func = nearest_neighbour_interpolation[np_floats, np_floats,

skimage/morphology/tests/test_skeletonize.py
72:    def test_skeletonize_num_neighbours(self):

skimage/morphology/tests/test_skeletonize_3d.py
108:def test_skeletonize_num_neighbours():

skimage/measure/tests/test_regionprops.py
394:    per = perimeter(SAMPLE.astype('double'), neighbourhood=8)

skimage/measure/_regionprops_utils.py
93:    geometry formula in discretized space. In practice, a neighbourhood
186:def perimeter(image, neighbourhood=4):
193:    neighbourhood : 4 or 8, optional
195:        compute the contour. A higher neighbourhood widens the border on which
216:    >>> perimeter(img_coins, neighbourhood=4)  # doctest: +ELLIPSIS
218:    >>> perimeter(img_coins, neighbourhood=8)  # doctest: +ELLIPSIS
225:    if neighbourhood == 4:

skimage/graph/_mcp.pyx
393:        from the heap, before examining the neighbours.
401:        the current point's neighbours and 2 if the algorithm is now
546:                    continue  # Skip neighbours
852:            The index of of the first neighbour in the connection.
854:            The index of of the second neighbour in the connection.

skimage/feature/_texture.pyx
99:        Number of circularly symmetric neighbour set points (quantization of
275:# Values represent offsets of neighbour rectangles relative to central one.

skimage/feature/texture.py
291:        Number of circularly symmetric neighbour set points (quantization of
466:    # Offsets of neighbour rectangles relative to central one.
468:    neighbour_rect_offsets = ((-1, -1), (-1, 0), (-1, 1),
473:    neighbour_rect_offsets = np.array(neighbour_rect_offsets)
474:    neighbour_rect_offsets[:, 0] *= height
475:    neighbour_rect_offsets[:, 1] *= width
481:    for element_num, offset in enumerate(neighbour_rect_offsets):

skimage/feature/_cascade.pyx
95:                                         int min_neighbour_number=4):
110:    min_neighbour_number : int
175:    clusters = threshold_clusters(clusters, min_neighbour_number)
653:                           min_neighbour_number=4,
678:        min_neighbour_number : int
784:                                      min_neighbour_number))

skimage/_shared/interpolation.pxd
42:cdef inline void nearest_neighbour_interpolation(
46:    """Nearest neighbour interpolation at a given position in the image.

doc/examples/segmentation/plot_perimeters.py
41:    # `neighbourhood` value can be 4 or 8 for the classic perimeter estimator

doc/examples/edges/plot_skeleton.py
48:# corresponding to each possible pattern of its 8 neighbouring

doc/examples/applications/plot_human_mitosis.py
133:# possibly, merging clusters of neighbouring objects (some could correspond to

doc/examples/applications/plot_face_detection.py
46:``min_neighbour_number`` and ``intersection_score_threshold`` parameters are
56:Then each cluster is thresholded using ``min_neighbour_number`` parameter which
```

**Comment by alexdesiqueira** [comment]
Alright, I guess we're good :slightly_smiling_face: 
```bash
❯ rg neighbour
skimage/measure/tests/test_regionprops.py
394:    with expected_warnings(["`neighbourhood` is a deprecated argument name"]):
395:        per = perimeter(SAMPLE.astype('double'), neighbourhood=8)

skimage/measure/_regionprops_utils.py
187:@deprecate_kwarg(kwarg_mapping={'neighbourhood': 'neighborhood'},

doc/release/release_dev.rst
29:- All references to EN-GB spelling for the word ``neighbour`` and others—e.g.,
31:  ``neighbor``. With that, ``skimage.measure.perimeter` parameter ``neighbourhood``
```
I think this is ready for your input, @scikit-image/core. Let me know what could be improved — especially the deprecation at `regionprops.perimeter`.

**Review by rfezzani — APPROVED** [review]
LGTM, thank you @alexdesiqueira :wink:

File: skimage/measure/_regionprops_utils.py [hunk 1]
```diff
@@ -183,16 +184,19 @@ def euler_number(image, connectivity=None):
         return int(0.125 * coefs @ h)
 
 
-def perimeter(image, neighbourhood=4):
+@deprecate_kwarg(kwarg_mapping={'neighbourhood': 'neighborhood'},
+                 removed_version="1.0",
+                 deprecated_version="0.19.1")
```
Comment (grlee77) [inline skimage/measure/_regionprops_utils.py:189]:
```suggestion
                 deprecated_version="0.19.2")
```

This was not yet deprecated in 0.19.1. I don't think we usually add deprecations in patch releases, either, though? We could just deprecate in the next release, but then removed version would be something like 1.2. I don't think we made a final decision on "1.0" vs. "2.0" though? (another reason to have a team meeting soon!)

File: doc/release/release_dev.rst [hunk 2]
```diff
@@ -26,7 +26,10 @@ Improvements
 API Changes
 -----------
 
-
+- All references to EN-GB spelling for the word ``neighbour`` and others—e.g.,
+  ``neigbourhood``, ``neighboring``, were changed to their EN-US spelling,
+  ``neighbor``. With that, ``skimage.measure.perimeter` parameter ``neighbourhood``
+  was deprecated in favor of ``neighborhood`` in 1.0.
```
Comment (grlee77) [inline doc/release/release_dev.rst:32]: This line indicates the deprecation is introduced in 1.0 which is inconsistent with the decorator which states that the argument is removed in 1.0.

**Review by grlee77 — CHANGES_REQUESTED** [review]
This looks good aside from deciding how best to handle the versions listed in the deprecation

**Comment by alexdesiqueira** [comment]
> This looks good aside from deciding how best to handle the versions listed in the deprecation

Thanks for these hints, @grlee77! It's the first time I'm using that decorator, so I'm kinda lost :smile:

**Comment by alexdesiqueira** [comment]
> It's the first time I'm using that decorator (...)

@grlee77 ... and I think it's the second time I'm actually deprecating anything :slightly_smiling_face:
Please let me know if this is correct now. Thanks again!

**Review by grlee77 — APPROVED** [review]
This looks good to me now, thanks @alexdesiqueira.

**Pull request merged**